src/cowFi-tokens.js

```diff
@@ -105,34 +117,29 @@ async function getIds() {
 
 async function getStaticData(idsData) {
   // If static data file exists, read and return it
-  if (fs.existsSync(STATIC_LIST_PATH_RAW)) {
+  if (USE_CACHE && fs.existsSync(STATIC_LIST_PATH_RAW)) {
     const json = fs.readFileSync(STATIC_LIST_PATH_RAW, "utf8");
     return JSON.parse(json);
   }
 
   const successDelay = 3000; // Delay in milliseconds for successful requests
-  const errorDelay = 10000; // Delay in milliseconds for error requests
   const staticData = [];
 
   let index = 0;
+  const totalTokens = idsData.length + 1;
   while (index < idsData.length) {
     const idItem = idsData[index];
     const quoteParams = `localization=false&tickers=false&market_data=true&community_data=false&developer_data=false&sparkline=false`;
     const url = `https://api.coingecko.com/api/v3/coins/${idItem.id}?${quoteParams}`;
 
-    console.log(`Fetching data for ID: ${idItem.id}`);
-
-    try {
-      const data = await fetchFromCoingecko(url);
-      staticData.push(data);
-      await sleep(successDelay); // Delay after a successful fetch
-      index++; // Move to the next item
-    } catch (error) {
-      console.log(error);
-      console.error(`Error fetching data for ID: ${idItem.id}`);
-      console.log(`Retrying fetch for ID: ${idItem.id}`);
-      await sleep(errorDelay); // Delay after an error fetch
-    }
+    console.log(
+      `[${index + 1}/${totalTokens}] Fetching data for ID: ${idItem.id}`
+    );
+
+    const data = await fetchWithBackoff(url);
+    staticData.push(data);
+    await sleep(successDelay); // Delay after a successful fetch
+    index++; // Move to the next item
```

Comment:
Since you removed `try...catch` here, is this going to move on to the next index if it fails, or is it going to keep fetching until its successful? Just want to check.